src/main/java/com/aws/greengrass/util/platforms/unix/UnixPlatform.java

```diff
@@ -355,20 +356,58 @@ public UnixRunWithGenerator getRunWithGenerator() {
 
     @Override
     public void createUser(String user) throws IOException {
-        runCmd("useradd -r -m " + user, o -> {
-        }, "Failed to create user");
+        boolean isExist = checkIfCommandExists("useradd");
+
+        if (isExist) {
+            runCmd("useradd -r -m " + user, o -> {
+            }, "Failed to create user");
+        } else {
+            runCmd("adduser -S  " + user, l -> {
+            }, "Failed to create user");
+        }
     }
 
     @Override
     public void createGroup(String group) throws IOException {
-        runCmd("groupadd -r " + group, o -> {
-        }, "Failed to create group");
+        boolean isExist = checkIfCommandExists("groupadd");
+
+        if (isExist) {
+            runCmd("groupadd -r " + group, o -> {
+            }, "Failed to create group");
+        } else {
+            runCmd("addgroup -S " + group, l -> {
+            }, "Failed to create group");
+        }
     }
 
     @Override
     public void addUserToGroup(String user, String group) throws IOException {
-        runCmd("usermod -a -G " + group + " " + user, o -> {
-        }, "Failed to add user to group");
+        boolean isExist = checkIfCommandExists("usermod");
+
+        if (isExist) {
+            runCmd("usermod -a -G " + group + " " + user, o -> {
+            }, "Failed to add user to group");
+        } else {
+            runCmd("addgroup " + user + " " + group, l -> {
+            }, "Failed to add user to group");
+        }
+    }
+
+    /**
+     * Check if command exists.
+     * @param command command
+     * @return boolean
+     * @throws IOException throws when io exception occurs
+     */
+    protected boolean checkIfCommandExists(String command) throws IOException {
+        AtomicBoolean flag = new AtomicBoolean(false);
+
+        // Determine whether this command exists
+        runCmd("command -v " + command + "|wc -l", o -> {
```

Comment:
You cannot rely on the command 'command' and 'wc' existing on the target platform. This has to work with no additional commands. You must determine what exists from entirely within Java.